cargo-dist/templates/installer/installer.sh.j2

```diff
@@ -307,7 +307,10 @@ install() {
     say "everything's installed!"
 
     if [ "0" = "$NO_MODIFY_PATH" ]; then
-        add_install_dir_to_path "$_install_dir_expr" "$_env_script_path" "$_env_script_path_expr"
+        local _rcfiles=".profile .bash_profile .bash_login .bashrc .zshrc .zshenv"
```

Comment:
I originally wrote this as an array, but shellcheck didn't let me since dash doesn't support arrays.

Reply:
So although it's still kind of a shotgun blast, iirc rustup is *slightly* more surgical than this. I believe for each kind of shell they identify families of files and write to the first one that already exists, or failing that write to a "default" location.

(Also some extra logic that looks in $ZDOTDIR in addition to $HOME...)

So like you'd write to:

* up to one file to provide base posix support
* up to one file to provide zsh support
* up to one file to provide bash support

For a maximum of 3 files written for total platform coverage.

https://github.com/rust-lang/rustup/blob/b5d74ac69307de29cee586c779bdb6b3aee7f7af/src/cli/self_update/shell.rs#L193-L206

Reply:
Updated. Now we write to three sets of scripts; we pick the first script we find that already exists from each set, or create the first listed script if none of them already exist.